tests/components/airly/test_sensor.py

```diff
@@ -41,7 +38,7 @@ async def test_sensor(hass, aioclient_mock):
     assert state.state == "23"
     assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
     assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "CAQI"
-    assert state.attributes.get(ATTR_DEVICE_CLASS) == DEVICE_CLASS_AQI
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.AQI
```

Comment:
I think it might make sense to use `is` rather than `==`

Reply:
Both are valid according to 

https://docs.python.org/3/library/enum.html#comparisons

But if there is a need/desire to to have consistency across all then yes I agree.

Reply:
The reason is that "string == enum" succeeds, but "string is enum" fails.
So it would be a way to ensure a string isn't inserted into the ATTR_DEVICE_CLASS by mistake

Reply:
Agreed. Thanks for explanation. 

Aside: could pylint or similar be configured to enforce this?

Reply:
We don't currently run pylint (or mypy for that matter) in `tests`.
https://github.com/home-assistant/core/blob/b03ead1c9b6355b1431e047c4005c2622a602b5e/.github/workflows/ci.yaml#L566-L577

Reply:
Ok. Using `is` instead of `==` generates pytest errors e.g.:

```
>       assert state.attributes.get(ATTR_DEVICE_CLASS) is SensorDeviceClass.HUMIDITY
E       AssertionError: assert 'humidity' is <SensorDeviceClass.HUMIDITY: 'humidity'>
E        +  where 'humidity' = <built-in method get of mappingproxy object at 0x7fa97f97bc40>('device_class')
E        +    where <built-in method get of mappingproxy object at 0x7fa97f97bc40> = mappingproxy({'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>, 'attribution': 'Data provided by Airly', 'unit_of_measurement': '%', 'device_class': 'humidity', 'friendly_name': 'Home Humidity'}).get
E        +      where mappingproxy({'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>, 'attribution': 'Data provided by Airly', 'unit_of_measurement': '%', 'device_class': 'humidity', 'friendly_name': 'Home Humidity'}) = <state sensor.home_humidity=92.8; state_class=measurement, attribution=Data provided by Airly, unit_of_measurement=%, device_class=humidity, friendly_name=Home Humidity @ 2021-12-16T11:06:06.562885+00:00>.attributes
E        +  and   <SensorDeviceClass.HUMIDITY: 'humidity'> = SensorDeviceClass.HUMIDITY

tests/components/airly/test_sensor.py:60: AssertionError
```

After a fair bit of digging I think I've got to the root cause.  If my understanding is correct, when pytest does the comparison it is looking at the state attributes rather than the entity attributes, and the state attributes are written by `Entity._async_write_ha_state()` which converts attributes to `str` here:

https://github.com/home-assistant/core/blob/a9879487cc5d7474ad6ccfd74fb03ae77c1b596d/homeassistant/helpers/entity.py#L575
If I change:
```
            attr[ATTR_DEVICE_CLASS] = str(device_class)
```
to
```
            attr[ATTR_DEVICE_CLASS] = device_class
```
Then the tests pass.

So (I think) the question @frenck is "should the state object be using string or enum representation of device class?"

- If it should be enum then the above edit should be made to homeassistant.helpers.entity.py:Entity()
- If it should be string, then we could solve it by casting to str during any comparisons in `test_*.py`.

Probably this should go into a new PR.

Reply:
Cc @scop

Reply:
States are written as strings.